### PR TITLE
[Inspection Service] Remove the Debug Service!

### PR DESCRIPTION
**Pull request by JoshLind**
### Description

This PR removes the `node debug service` and renames the `metrics service` to `inspection service`. Instead of exposing two services (and ports), we now have a single unified service that offers:
- `/metrics` -> the metrics endpoint exposing human readable metrics.
- `/system_information` -> system and build information for the node.
- `/json_metrics` and `/forge_metrics` -> metrics encoded in different ways for different use cases. It's still TBD if anyone is actually scraping `json_metrics` though. But, we'll keep it for now as it's easy for a client to parse.
- `/configuration` -> the configuration endpoint exposing the node configuration. This is disabled by default to prevent sensitive information from being mistakenly leaked (e.g., file paths and/or keys).

Note:
- We've removed the periodic state dump that logs the node config every hour and the synced version every minute. It's unclear if this was ever used. The logs and metrics should be sufficient.
- We've removed the ability to update the logger dynamically. Again, I've never seen it used (and in production we expect this port to be closed anyway).
- We've added a new `InspectionServiceConfig` allowing the `/configuration` and `/system_information` endpoints to be disabled.

### Test Plan

Existing unit tests and manual verification.

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/1439)

### CODE REVIEW

File: config/src/config/inspection_service_config.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::utils;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct InspectionServiceConfig {
```
Comment (davidiw) [inline config/src/config/inspection_service_config.rs:9]: Why not Metrics?

**Review by davidiw — COMMENTED** [review]
InspectionServer seems like an awkward name. I don't know if folks that want metrics will know where to find it.

File: config/src/config/inspection_service_config.rs (hunk 1 above)
Comment (JoshLind) [reply config/src/config/inspection_service_config.rs:9]: This is already more than just metrics (e.g., system info, build info and node configuration) and I think we'll likely want to extend it further in the future (e.g., interactive debugging, deep introspection etc.).

Comment (davidiw) [reply config/src/config/inspection_service_config.rs:9]: Maybe debug was a better word lol ... NodeStatus / NodeState / Metrics!

**Review by davidiw — APPROVED** [review]

**Pull request merged**